cmd/snap-confine/snap-confine.apparmor.in

```diff
@@ -404,4 +382,116 @@
+    # various locations of libexecdir across distributions.
+    /usr/lib{,exec}/snapd/snap-update-ns Cxr -> snap_update_ns,
+
+    # ...snap-confine is not, conceptually, re-executing and uses
+    # snap-update-ns from the distribution package but we are already inside
+    # the constructed mount namespace so we must traverse "hostfs". The
+    # variants here represent various locations of libexecdir across
+    # distributions.
+    /var/lib/snapd/hostfs/usr/lib{,exec}/snapd/snap-update-ns Cxr -> snap_update_ns,
+
+    # ..snap-confine is, conceptually, re-executing and uses snap-update-ns
+    # from the core snap. Note that the location of the core snap varies from
+    # distribution to distribution. The variants here represent different
+    # locations of snap mount directory across distributions.
+    /{,var/lib/snapd/}snap/core/*/usr/lib/snapd/snap-update-ns Cxr -> snap_update_ns,
+
+    # ...snap-confine is, conceptually, re-executing and uses snap-update-ns
+    # from the core snap but we are already inside the constructed mount
+    # namespace. Here the apparmor kernel module re-constructs the path to
+    # snap-update-ns using the "hostfs" mount entry rather than the more
+    # "natural" /snap mount entry but we have no control over that.  This is
+    # reported as (LP: #1716339). The variants here represent different
+    # locations of snap mount directory across distributions.
+    /var/lib/snapd/hostfs/{,var/lib/snapd/}snap/core/*/usr/lib/snapd/snap-update-ns Cxr -> snap_update_ns,
+
+    profile snap_update_ns (attach_disconnected) {
+        # The next four rules mirror those above. We want to be able to read
+        # and map snap-update-ns into memory but it may come from a variety of places.
+        /usr/lib{,exec}/snapd/snap-update-ns mr,
+        /var/lib/snapd/hostfs/usr/lib{,exec}/snapd/snap-update-ns mr,
+        /{,var/lib/snapd/}snap/core/*/usr/lib/snapd/snap-update-ns mr,
+        /var/lib/snapd/hostfs/{,var/lib/snapd/}snap/core/*/usr/lib/snapd/snap-update-ns mr,
+
+        # Allow reading the dynamic linker cache.
+        /etc/ld.so.cache r,
+        # Allow reading, mapping and executing the dynamic linker.
+        /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}ld-*.so mrix,
+        # Allow reading and mapping various parts of the standard library and
+        # dynamically loaded nss modules and what not.
+        /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libc{,-[0-9]*}.so* mr,
+        /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libdl{,-[0-9]*}.so* mr,
+        /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libgcc_s.so* mr,
+        /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libpcre.so* mr,
+        /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libpthread{,-[0-9]*}.so* mr,
+        /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libresolv{,-[0-9]*}.so* mr,
+        /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}librt{,-[0-9]*}.so* mr,
+        /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libselinux.so* mr,
```

Comment:
Done though I suspect it will be different than this on Solus. In any case before we have Solus CI have reduced those. I'll send distro-specific PRs if this is insufficient.